Source/Core/Common/FileUtil.h

```diff
@@ -211,7 +211,7 @@ class IOFile : public NonCopyable
   bool IsOpen() const { return nullptr != m_file; }
   // m_good is set to false when a read, write or other function fails
   bool IsGood() const { return m_good; }
-  operator void*() { return m_good ? m_file : nullptr; }
+  explicit operator bool() const { return IsGood(); }
```

Comment:
This changes the semantic of the conversion: before it would be true if m_good && m_file. Now it can be true even if m_file is not a true value.